src/main/java/picocli/CommandLine.java

```diff
@@ -4799,6 +4804,22 @@ private static String format(String formatString, Object... params) {
         public String[] getVersion() throws Exception { throw new UnsupportedOperationException(); }
     }
 
+    /**
+     * Provides a way to modify how command model is built. Commands may configure a preprocessor using
+     * {@link Command#modelTransformer()} annotation attribute.
+     * @since X.X */
+    public interface IModelTransformer {
+        /**
+         * Returns CommandSpec after doing preprocessing.
+         * @return preprocessed CommandSpec
+         */
+        CommandSpec transform(CommandSpec commandSpec);
+    }
+
+    private static class NoOpPreprocessor implements IModelTransformer {
```

Comment:
Let's make this NoOpModelTransformer too.

Reply:
Sure.